apps/hash-frontend/src/pages/drafts.page/draft-entity.tsx

```diff
@@ -40,14 +56,51 @@ export const DraftEntity: FunctionComponent<{
     [subgraph, entity],
   );
 
+  const [entityRootedSubgraph, setEntityRootedSubgraph] = useState<
+    Subgraph<EntityRootType>
+  >(generateEntityRootedSubgraph(entity, subgraph));
+
+  const previouslyEvaluatedEntity = useRef<Entity>(entity);
+
+  /**
+   * Only re-evaluate the entity rooted subgraph if the entity edition
+   * has changed, as otherwise the entity editor entity selector will re-mount
+   * itself clearing any text input from the user. Ideally this would not be
+   * the case, and the latest version of the subgraph would become available
+   * to the entity editor, so that we can do something like this:
+   *
+   * const entityRootedSubgraph = useMemo<Subgraph<EntityRootType>>(
+   *   () => generateEntityRootedSubgraph(entity, subgraph),
+   *   [subgraph, entity],
+   * );
+   *
+   * @todo: figure out what the underlying issue is causing the `EntitySelector`
+   * component to re-mount when the subgraph changes.
+   */
+  if (
+    previouslyEvaluatedEntity.current.metadata.recordId.editionId !==
+    entity.metadata.recordId.editionId
+  ) {
+    previouslyEvaluatedEntity.current = entity;
+    setEntityRootedSubgraph(generateEntityRootedSubgraph(entity, subgraph));
+  }
```

Comment:
@CiaranMn I spent a significant amount of time trying to figure out the root cause of this, as it should be addressed in the entity editor instead of here. This also unfortunately doesn't fully resolve the issue, as when a property is updated in the entity editor there is are updates to the `entitySubgraph` which cause the entity selector to re-mount. This is obvious when updating a property, and in quick succession trying to add a link:

https://github.com/hashintel/hash/assets/42802102/2341dcf6-427f-4656-9a2c-d636667c99de

I'm quite confused by what's causing the re-mount as the text fields in the property editor are not affected by this, and it's not happening on the dedicated entity editor page. This means it must be something specific to how the subgraph is passed to the entity editor in the `EditEntityModal`, I just haven't been able to think of what that might be.

Reply:
My theory (might be wrong) is that it could be because you use `entity` as a dependency to the intended approach, which causes the subgraph to be recalculate whenever the object's identity changes. Using `entityId` as the dependency instead might fix it
```ts
   * const entityRootedSubgraph = useMemo<Subgraph<EntityRootType>>(
   *   () => generateEntityRootedSubgraph(entity, subgraph),
   *   [subgraph, entity],
   * );
 ```

Reply:
I've tried this and it does not fix the issue. Separately I've tested selecting files in the entity selector to verify the subgraph changes still enable previewing/linking to those entities:

<img width="1335" alt="image" src="https://github.com/hashintel/hash/assets/42802102/92634ad2-a138-4f9e-a381-252717a45100">

Reply:
Cool, thanks Ben – we talked about this and I made a mistake in my view above, because what you're doing isn't the same as making `entityId` a dependency of the original approach. It's excluding `subgraph` as a dependency of recalculating subgraph. But if things still seem to work let's merge it for now and figure out the issue with the selector in follow-up